airflow/api_fastapi/core_api/datamodels/assets.py

```diff
@@ -64,3 +64,45 @@ class AssetCollectionResponse(BaseModel):
 
     assets: list[AssetResponse]
     total_entries: int
+
+
+class DagRunAssetReference(BaseModel):
+    """Serializable version of the DagRunAssetReference ORM SqlAlchemyModel."""
+
+    run_id: str
+    dag_id: str
+    execution_date: datetime = Field(alias="logical_date")
+    start_date: datetime
+    end_date: datetime | None = None
+    state: str
+    data_interval_start: datetime
+    data_interval_end: datetime
+
+
+class AssetEventResponse(BaseModel):
+    """Asset event serializer for responses."""
+
+    id: int
+    asset_id: int
+    asset_uri: str
+    extra: dict | None = None
+    source_task_id: str | None = None
+    source_dag_id: str | None = None
+    source_run_id: str | None = None
+    source_map_index: int
+    created_dagruns: list[DagRunAssetReference]
+    timestamp: datetime
+
+    @model_validator(mode="before")
+    def rename_uri_to_asset_uri(cls, values):
+        """Rename 'uri' to 'asset_uri' during serialization to match legacy response."""
+        if hasattr(values, "uri") and values.uri:
+            values.asset_uri = values.uri
+        return values
```

Comment:
Aliases can achieve that in a cleaner way I believe. (`serialization_alias`, `validation_alias`, etc... there are a lot of option from pydantic to convert from python to json world with different names).

We have several examples in the code for different type of aliases, I think you might find some of interest for this use case.

Reply:
I checked this thanks.

This PR has some duplicate effort from this one https://github.com/apache/airflow/pull/43881
It makes sense to me that since we are introducing a breaking change anyways, I will change this to be `uri` as per this comment https://github.com/apache/airflow/pull/43881

Reply:
Made the change in the latest commit